Radio documentation: updating radio readme

davewhitley opened the pull request:
Initial commit to update Radio readme. These changes add design documentation.

## Description
<!-- Please describe what you have changed or added -->

This PR adds design guideline documentation in addition to the development documentation that existed previously. These guidelines are 'best practices' for the usage of the component, as well as describing the component in more detail.

## Screenshots <!-- if applicable -->

A preview can be seen here, although it doesn't represent what it will look like in the handbook:

http://simp.ly/p/qzstJq

## Feedback

I'd appreciate feedback on:

- the image alt text
- usage of the `h1` header 
- design guidelines
- finding a way to preview these changes and see what they'll look like in the handbook

Thank you!

davewhitley committed:
Updating Radio readme

Initial commit to update Radio readme. These changes add design documentation.

davewhitley committed:
Adding readme images

chrisvanpatten commented on packages/components/src/radio-control/README.md:
```diff
@@ -1,11 +1,59 @@
+2. [Development guidelines](#development-guidelines)
+3. [Related components](#related-components)
+
+# Design guidelines
```
It's such a small thing but it would be good to get an accessibility review on this; I know in the past it was suggested to keep one `<h1>` per page, but that may have changed. I know some a11y tech uses heading levels to assist navigation so it's important to get this right.

chrisvanpatten commented on packages/components/src/radio-control/README.md:
```diff
@@ -1,11 +1,59 @@
+
+If you have a list of available options can be collapsed, consider using a dropdown menu instead — it uses less space.
+
+![Use radio buttons when only one item can be selected from a list.](./radio-usage-do.png)
```
I'm not sure the `alt` attribute text should duplicate the text label. My understanding is that `alt` should describe what users would "see" in the image, but this text sort of augments the image without clearly explaining what the image depicts.

chrisvanpatten commented on packages/components/src/radio-control/README.md:
```diff
@@ -1,11 +1,59 @@
+
+### Defaults
+
+When using radio buttons **one must be selected by default** (i.e., when the page loads, in the case of a web application).
```
I'm not sure this is true. I know it's recommended, but we don't enforce it in the component and it is possible that a user could program a "reset" button which allows a radio list to be in a "null" state without a selection.

If we're just recommending this as a "best practices", "must" could be replaced with "should".

davewhitley committed:
Image path update

Updating the path of the images. Making a small tweak to phrasing.

drw158 commented:
Thanks for the review @chrisvanpatten. I would love some a11y guidance on the alt text and use of headers.

Sounds like next steps for this would be:

- [ ] Update images to be consistent with Joen's new do/don't images
- [ ] Determine best way forward for alt text on these images
- [ ] Figure out if there should only be a single `h1` on the page

drw158 commented:
I just read this to get a better handle on alt text and captions: https://webaim.org/techniques/alttext/

I'm still not sure, but leaving the alt text blank might be a valid approach because of the surrounding text.

Another possibility is describing what is going on in the diagram. So, for example, this image:

![screen shot 2018-10-30 at 11 27 55 am](https://user-images.githubusercontent.com/618551/47733733-df72fd80-dc36-11e8-8fd1-56db1ffd03ce.png)

the alt text could be:

"A diagram of an interface using checkboxes when only one item can be selected from a list."

davidakennedy commented:
> I would love some a11y guidance on the alt text and use of headers.

Hi @drw158! Happy to share some guidance. Love that you're thinking about this, and asking great questions. ❤️ 

First, for the headings:

- Ideally, there should be one h1 on a page. Why? Because screen reader users often scan a page by heading levels, much like sighted users skim a page visually looking for the headlines. Having one h1 helps them know – this is the page title, the topic of the page. An added benefit is SEO. One h1 has some benefits there as well.
- Even though HTML5 supports a document outline that can include multiple h1s in a page, it's not supported by any browser, so it's not practical to rely on.

For the alt attributes:
- What I do in this case is imagine that the images aren't there. Would the content lose any value? In this case, no. As you say, the content covers the same guidelines that the second and third images show. It's fine to use a blank alt attribute on those.

samikeijonen commented:
I agree with @davidakennedy in the [previous comment](https://github.com/WordPress/gutenberg/pull/11226#issuecomment-437421535).

Although I can't really see the context of the image and surrounding text. If alt is needed it could be like "Post visibility checkboxes: public, private, and password protected." 

@drw158 Any change to update the PR?

davewhitley committed:
Updates so that only a single h1 is present

davewhitley committed:
image alt text removal

The alt text for these images has been removed because removing the images from the document doesn't affect comprehension.

drw158 commented:
@samikeijonen Can you explain what you mean when you say you can't see the context of the image?

![screen shot 2018-11-28 at 2 56 10 pm](https://user-images.githubusercontent.com/618551/49181813-fda74880-f31d-11e8-8c8f-3830ca30d251.png)

@chrisvanpatten all of your concerns should be fixed

---

I'm not sure where the images should be uploaded. That is the only blocker right now.

davewhitley committed:
Removing images

davewhitley committed:
Removing images

davewhitley committed:
Removing images

davewhitley committed:
Adding images hosted on wordpress.org

drw158 commented:
Image issue has been resolved. Now ready for a final review and merge.

tofumatt committed:
docs: tweak some usage recommendations

tofumatt committed:
Tweak spacing

tofumatt commented on packages/components/src/radio-control/README.md:
```diff
@@ -1,11 +1,59 @@
+
+When using radio buttons **one should be selected by default** (i.e., when the page loads, in the case of a web application).
+
+**User control**
```
These are headers, not paragraph text; they should be labelled with header markdown (eg `##### User control`), not bold text like this. 😄

tofumatt committed:
docs: Use headers

tofumatt reviewed (approved):
This is an improvement so I'm gonna say let's go for it.

chrisvanpatten reviewed (approved):
Looks good to me and I'm glad we're using make/design for image uploads!
